org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/syntaxcoloring/SemanticHighlightingCalculator.java

```diff
@@ -104,4 +105,13 @@ public INode getFirstFeatureNode(EObject semantic, EStructuralFeature feature) {
 			return nodes.get(0);
 		return null;
 	}
+	
+	@Override
+	protected boolean highlightElement(EObject object, IHighlightedPositionAcceptor acceptor, CancelIndicator cancelIndicator) {
```

Comment:
Could you shift this into an else block for the condition in line 92? e.g. `if (SPECIAL...) .. else { default }` but *only* if the text is `true` or `false` to reduce the number of emitted highlighting tokens?

Reply:
Done.